luarules/gadgets/game_no_rush_mode.lua

```diff
@@ -73,16 +73,20 @@ if gadgetHandler:IsSyncedCode() then
 		if frame < norushtimer and (not TeamIDsToExclude[unitTeam]) then
 			local _, _, _, _, _, allyTeamID = Spring.GetTeamInfo(unitTeam)
 			if cmdID == CMD.INSERT then
+			  -- CMD.INSERT wraps another command, so we need to extract it
 			  cmdID = cmdParams[2]
-			  local s = #cmdParams == 7 and 4 or 3
-			  local e = #cmdParams
-			  for i = 1, s do
-				cmdParams[i] = cmdParams[i + 3]
+			  -- Area commands have an extra radius parameter, so they shift by 4 instead of 3
+			  local isAreaCommand = (#cmdParams == 7)
+			  local paramCountToShift = isAreaCommand and 4 or 3
```

Comment:
Commands can have an arbitrary number of params.
```suggestion
			  local paramCountToShift = #cmdParams - 3
```